app/tree/gui/src/ts/twxplore/gui/tree/components/map/Map.tsx

```diff
@@ -0,0 +1,130 @@
+    const dispatch = useDispatch();
+    const state: MapState = useSelector((rootState: RootState) => rootState.app.map);
+    // Load boroughs on first render
+    const boroughsQueryResult = useQuery<BoroughsQuery>(boroughsQueryDocument, {});
+    
+    const [getNtasByBorough,{}] = useLazyQuery<NtasByBoroughQuery, NtasByBoroughQueryVariables>(ntasByBoroughQueryDocument, {
+        onCompleted: (data: NtasByBoroughQuery) => { console.log("this is my story")
+            dispatch(addMapFeatures(data.ntas.byBoroughGeometry.map(ntaFeature => ({
+            childType: MapFeatureType.BLOCKFACE,
+            geometry: ntaFeature.geometry,
+            state: MapFeatureState.LOADED,
+            type: MapFeatureType.NTA,
+            uri: ntaFeature.uri
+        }))))}
+        
+    })
+
+    if (state.features.length === 0) {
+        if (boroughsQueryResult.data) {
+            // No tracking any features yet, add the boroughs we loaded
+            dispatch(addMapFeatures(boroughsQueryResult.data.boroughs.geometries.map(boroughFeature => ({
+                childType: MapFeatureType.NTA,
+                geometry: boroughFeature.geometry,
+                state: MapFeatureState.LOADED,
+                type: MapFeatureType.BOROUGH,
+                uri: boroughFeature.uri
+            }))))
+            console.log(state.features.length)
+        }
+    }
+
+    // Organize the features by state
+    const featuresByState: { [index: string]: MapFeature[] } = {};
+    for (const feature of state.features) {
+        const features = featuresByState[feature.state];
+        if (features) {
+            features.push(feature);
+        } else {
+            featuresByState[feature.state] = [feature];
+        }
+        console.log(featuresByState)
+    }
+
+    // Feature state machine
+    for (const featureState in featuresByState) {
+        const featuresInState = featuresByState[featureState];
+        switch (featureState) {
+            case MapFeatureState.LOADED: {
+                const datasets = {
+                    data: Processors.processGeojson({
+                        "type": "FeatureCollection",
+                        "features": featuresInState.map(feature => {
+                            return {
+                                "type": "Feature",
+                                "geometry": wkt.parse(feature.geometry.wkt),
+                                "properties": feature
+                            }
+                        })
+                    }),
+                    info: {
+                        id: featuresInState[0].type
+                    }
+                }
+                dispatch(addDataToMap({datasets, options: {centerMap: true, readOnly: true}}))
+                break;
+            }
+            case MapFeatureState.CLICKED: {
+                for (const clickedFeature of featuresInState){
+                 getNtasByBorough({variables: {uri: clickedFeature.uri}})
```

Comment:
I filed an issue to set up a tool to get us consistent formatting. Will fix that soon.